libtelemetry/src/full/java/com/mapbox/android/telemetry/MapboxTelemetry.java

```diff
@@ -479,8 +479,8 @@ public Thread newThread(Runnable runnable) {
   }
 
   @SuppressWarnings("WeakerAccess")
-  public void setBaseUrl(String eventsHost) {
-    if (isValidUrl(eventsHost)) {
+  public synchronized void setBaseUrl(String eventsHost) {
```

Comment:
I wonder if just ignoring this call is ok, if network is not connected at the moment of the call

Reply:
network check can be removed, so we can call
`checkRequiredParameters(sAccessToken.get(), userAgent)`

Reply:
@korshaknn maybe we can return result if base url was set successfully?
I'm afraid with current approach it can be dangerous setBaseUrl was called and ignored and then telemetry will be send to the wrong host